pkg/kine/drivers/sqlite/sqlite.go

```diff
@@ -148,34 +153,43 @@ func setup(ctx context.Context, db *sql.DB) error {
 // migrate tries to migrate from a version of the database
 // to the target one.
 func migrate(ctx context.Context, txn *sql.Tx) error {
-	var userVersion int
+	var userVersion int32
 
 	row := txn.QueryRowContext(ctx, `PRAGMA user_version`)
 	if err := row.Scan(&userVersion); err != nil {
 		return err
 	}
+	currentMajor, currentMinor := getMajorMinorVersion(int32(userVersion))
 
-	switch userVersion {
-	case 0:
+	if currentMajor == 0 && currentMinor == 0 {
 		if err := applySchemaV1(ctx, txn); err != nil {
 			return err
 		}
-		fallthrough
-	case databaseSchemaVersion:
-		break
-	default:
-		// FIXME this needs better handling
-		return errors.Errorf("unsupported version: %d", userVersion)
 	}
 
-	setUserVersionSQL := fmt.Sprintf(`PRAGMA user_version = %d`, databaseSchemaVersion)
+	setUserVersionSQL := fmt.Sprintf(`PRAGMA user_version = %d`, toInt32(databaseSchemaMajorVersion, databaseSchemaMinorVersion))
 	if _, err := txn.ExecContext(ctx, setUserVersionSQL); err != nil {
 		return err
 	}
 
 	return nil
 }
 
+func getMajorMinorVersion(value int32) (major int16, minor int16) {
```

Comment:
Can we have a type for the version? Something like:

```go
type schemaVersion int

func (sv schemaVersion) Major() int
func (sv schemaVersion) Minor() int

func MakeSchemaVersion(major, minor int) schemaVersion
```

You could also add a method like:

```go
func (sv schemaVersion)  CompatibleWith(other schemaVersion) error
```

Which checks if two schema version are compatible.